wiremock-webhooks-extension/src/test/java/functional/WebhooksAcceptanceTest.java

```diff
@@ -199,22 +202,30 @@ public void webhookCanBeConfiguredFromJson() throws Exception {
     verify(postRequestedFor(urlPathEqualTo("/callback2")));
   }
 
-  @Test
-  public void appliesTemplatingToUrlMethodHeadersAndBodyViaDSL() throws Exception {
+  private WebhookDefinition withBodyOrBodyFileName(String bodyFileName, WebhookDefinition webhook) {
+      if (!bodyFileName.isEmpty()) { 
+          webhook.withBodyFileName(bodyFileName);
+      }
+      return webhook;
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"", "myFile.json"})
+  public void appliesTemplatingToUrlMethodHeadersAndBodyViaDSL(String bodyFileNameParam) throws Exception {
     rule.stubFor(
         post(urlPathEqualTo("/templating"))
             .willReturn(ok())
             .withPostServeAction(
                 "webhook",
-                webhook()
+                withBodyOrBodyFileName(bodyFileNameParam, webhook()
                     .withMethod("{{jsonPath originalRequest.body '$.method'}}")
                     .withUrl(
                         targetServer.baseUrl()
                             + "{{{jsonPath originalRequest.body '$.callbackPath'}}}")
                     .withHeader("X-Single", "{{math 1 '+' 2}}")
                     .withHeader("X-Multi", "{{math 3 'x' 2}}", "{{parameters.one}}")
                     .withBody("{{jsonPath originalRequest.body '$.name'}}")
```

Comment:
I feel like specifying both the body and the body file is a bit confusing here. Perhaps 2 separate tests would be better rather than one parameterised one?

Reply:
Yes - I have duplicated the tests now, one java/DSL set for bodyFileName and one set for body as a string.